compiler/rustc_mir/src/borrow_check/diagnostics/conflict_errors.rs

```diff
@@ -1543,6 +1542,37 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
             None,
         );
 
+        let tcx = self.infcx.tcx;
+        // point out implicit deref coercion
+        if let (
+            Some(Terminator { kind: TerminatorKind::Call { from_hir_call, .. }, .. }),
+            Some((method_did, method_substs)),
+        ) = (
+            &self.body[loan.reserve_location.block].terminator,
+            crate::util::find_self_call(
+                tcx,
+                &self.body,
+                loan.assigned_place.local,
+                loan.reserve_location.block,
+            ),
+        ) {
+            let is_deref = !from_hir_call && tcx.is_diagnostic_item(sym::deref_method, method_did);
```

Comment:
Just out of interest: can you explain why we only want to output this error note if `from_hir_call` is false?

Reply:
I would like to add the note in the output for something like `issue-81365.rs` where the deref is implicit, v. something like `issue-81365-8.rs`, where the deref is explicit and shouldn't be added. In fact, it is what I'm working on now, trying to figure out if the deref was explicit or not.

Reply:
Actually, it seems like `from_hir_call` is true for a syntactic call like `f(...)` or `x.f()` but false for an operator, perhaps overloaded, like `*x` or the implicitly inserted deref in something like `(&x).0`.